sdks/wasm/DebuggerTestSuite/DateTimeTests.cs

```diff
@@ -0,0 +1,375 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+using System.Net.WebSockets;
+using System.Threading;
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+
+using Newtonsoft.Json.Linq;
+using Xunit;
+using WebAssembly.Net.Debugging;
+using System.Globalization;
+
+namespace DebuggerTests
+{
+	public class DateTimeList : DebuggerTestBase {
+		DebugTestContext ctx;
+		Dictionary<string, string> dicScriptsIdToUrl;
+		Dictionary<string, string> dicFileToUrl;
+		Dictionary<string, string> SubscribeToScripts (Inspector insp) {
+			dicScriptsIdToUrl = new Dictionary<string, string> ();
+			dicFileToUrl = new Dictionary<string, string>();
+			insp.On("Debugger.scriptParsed", async (args, c) => {
+				var script_id = args? ["scriptId"]?.Value<string> ();
+				var url = args["url"]?.Value<string> ();
+				if (script_id.StartsWith("dotnet://"))
+				{
+					var dbgUrl = args["dotNetUrl"]?.Value<string>();
+					var arrStr = dbgUrl.Split("/");
+					dbgUrl = arrStr[0] + "/" + arrStr[1] + "/" + arrStr[2] + "/" + arrStr[arrStr.Length - 1];
+					dicScriptsIdToUrl[script_id] = dbgUrl;
+					dicFileToUrl[dbgUrl] = args["url"]?.Value<string>();
+				} else if (!String.IsNullOrEmpty (url)) {
+					dicFileToUrl[new Uri (url).AbsolutePath] = url;
+				}
+				await Task.FromResult (0);
+			});
+			return dicScriptsIdToUrl;
+		}
+		public DateTimeList() : base ("debugger-locale-driver.html") {}
+
+		[Fact]
+		public async Task CheckThatAllLocaleSourcesAreSent () {
+			var insp = new Inspector ();
+			//Collect events
+			var scripts = SubscribeToScripts(insp);
+
+			await Ready();
+			//all sources are sent before runtime ready is sent, nothing to check
+			await insp.Ready ();
+			Assert.Contains ("dotnet://debugger-test.dll/debugger-datetime-test.cs", scripts.Values);
+		}
+
+		void CheckLocation (string script_loc, int line, int column, Dictionary<string, string> scripts, JToken location)
```

Comment:
You can move all the common code to a new base class derived from `DebuggerTestBase`, or derive this class from `SourceList`.

Reply:
If the new driver isn't required, then these tests can be moved to be with the existing tests?

Reply:
New driver not required but @lewing suggested creating a new driver since the old driver was getting crowded. I do agree that some of the helper methods can just be moved to the original `DebuggerTestBase`

Reply:
We are not adding any new functions to the driver now, because we have `invoke_static_method`. We could even update the existing tests to remove:

```
this.int_add = Module.mono_bind_static_method ("[debugger-test] Math:IntAdd");
this.use_complex = Module.mono_bind_static_method ("[debugger-test] Math:UseComplex");
this.delegates_test = Module.mono_bind_static_method ("[debugger-test] Math:DelegatesTest");
this.generic_types_test = Module.mono_bind_static_method ("[debugger-test] Math:GenericTypesTest");
this.outer_method = Module.mono_bind_static_method ("[debugger-test] Math:OuterMethod");
this.async_method = Module.mono_bind_static_method ("[debugger-test] Math/NestedInMath:AsyncTest");
this.method_with_structs = Module.mono_bind_static_method ("[debugger-test] DebuggerTests.ValueTypesTest:MethodWithLocalStructs");
```
.. and use `invoke_static_method`. I'll leave it to you, and @lewing to decide :)

Reply:
Yes, I've been using `invoke_static_method` for the new driver as well.